extensions/amp-list/0.1/amp-list.js

```diff
@@ -179,21 +194,41 @@ export class AmpList extends AMP.BaseElement {
       if (maxLen < items.length) {
         items = items.slice(0, maxLen);
       }
-      return this.renderItems_(items);
+      return this.scheduleRender_(items);
     }, error => {
       throw user().createError('Error fetching amp-list', error);
     });
   }
 
   /**
+   * Schedules a fetch result to be rendered in the near future.
    * @param {!Array} items
    * @return {!Promise}
    * @private
    */
-  renderItems_(items) {
-    return this.templates_.findAndRenderTemplateArray(this.element, items)
+  scheduleRender_(items) {
+    let resolver;
+    let rejecter;
+    const promise = new Promise((resolve, reject) => {
+      resolver = resolve;
+      rejecter = reject;
+    });
+    this.renderQueue_.push({items, resolver, rejecter});
+    this.renderPass_.schedule();
+    return promise;
+  }
+
+  /**
+   * Dequeues a fetch result and renders it immediately.
+   * @private
+   */
+  doRenderPass_() {
+    dev().assert(this.renderQueue_.length > 0, 'Nothing queued to render.');
+    const {items, resolver, rejecter} = this.renderQueue_.shift();
```

Comment:
This is likely a memory leak, there's nothing stopping `renderQueue_` from being multiple items. These multiple items get queued, but the first (and rest) won't process until the next time a render pass is scheduled.

Reply:
:feelsgood: #13729